app/src/org/commcare/fragments/personalId/PersonalIdPhoneFragment.java

```diff
@@ -188,8 +232,97 @@ private void startConfigurationRequest() {
         });
     }
 
+    @Override
+    public void onLocationResult(@NonNull Location result) {
+        location = result;
+        updateContinueButtonState();
+    }
+
+    @Override
+    public void onLocationRequestFailure(@NonNull Failure failure) {
+        if (failure instanceof CommCareLocationListener.Failure.ApiException) {
+            Exception exception = ((CommCareLocationListener.Failure.ApiException)failure).getException();
+            if (exception instanceof ResolvableApiException) {
+                handleNoLocationServiceProviders();
+            }
+        }
+    }
+
+    private void handleNoLocationServiceProviders() {
+        DialogInterface.OnCancelListener onCancelListener = dialog -> {
+            location = null;
+            navigateToPermissionErrorMessageDisplay(R.string.personalid_location_service_error,
+                    R.string.personalid_grant_location_service);
+        };
+
+        DialogInterface.OnClickListener onChangeListener = (dialog, i) -> {
+            switch (i) {
+                case DialogInterface.BUTTON_POSITIVE:
+                    GeoUtils.goToProperLocationSettingsScreen((AppCompatActivity)requireActivity());
+                    break;
+                case DialogInterface.BUTTON_NEGATIVE:
+                    location = null;
+                    navigateToPermissionErrorMessageDisplay(R.string.personalid_location_service_error,
+                            R.string.personalid_grant_location_service);
+                    break;
+            }
+            dialog.dismiss();
+        };
+
+        GeoUtils.showNoGpsDialog((AppCompatActivity)requireActivity(), onChangeListener, onCancelListener);
+    }
+
+    @Override
+    public void onLocationRequestStart() {
+    }
+
+    @Override
+    public void requestNeededPermissions(int requestCode) {
```

Comment:
no reason to implement `RuntimePermissionRequester` for this class if we are not using `requestNeededPermissions`